### PR TITLE
Fix inconsistencies in GrpcClientTest expectations.

### PR DESCRIPTION
**Pull request by swankjesse**
Also fix MockRouteGuideService to not mask test crashes as timeouts.

### CODE REVIEW

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt [hunk 1]
```diff
@@ -563,10 +567,11 @@ class GrpcClientTest {
     }
   }
 
-  @Test @Ignore("Flaky")
+  @Test
   fun duplexBlockingReceiveOnly() {
     mockService.enqueue(ReceiveCall("/routeguide.RouteGuide/RouteChat"))
     mockService.enqueueSendNote(message = "welcome")
+    mockService.enqueueSendNote(message = "polo")
```
Comment (swankjesse) [inline wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt]: we were expecting to read this below

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt [hunk 2]
```diff
@@ -81,6 +81,11 @@ class GrpcClientTest {
     routeGuideService = grpcClient.create(RouteGuideClient::class)
   }
 
+  @After
+  fun tearDown() {
+    okhttpClient.dispatcher.executorService.shutdown()
```
Comment (swankjesse) [inline wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt]: To tidy up threads after the test completes

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt [hunk 3]
```diff
@@ -700,10 +704,10 @@ class GrpcClientTest {
   }
 
   @Test
-  @Ignore("Timing out")
   fun grpcStreamingCallIsExecutedAfterExecuteBlocking() {
     mockService.enqueue(ReceiveCall("/routeguide.RouteGuide/RouteChat"))
     mockService.enqueueSendNote(message = "welcome")
+    mockService.enqueueSendNote(message = "polo")
```
Comment (swankjesse) [inline wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt]: we expect to receive this below

File: wire-grpc-tests/src/test/java/com/squareup/wire/MockRouteGuideService.kt [hunk 4]
```diff
@@ -116,8 +116,8 @@ class MockRouteGuideService : RouteGuideGrpc.RouteGuideImplBase(), TestRule {
           base.evaluate()
         } finally {
           server.shutdown()
-          server.awaitTermination()
         }
+        server.awaitTermination()
```
Comment (swankjesse) [inline wire-grpc-tests/src/test/java/com/squareup/wire/MockRouteGuideService.kt]: we were doing this even when the test throws, which didn't make sense

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt [hunk 5]
```diff
@@ -32,9 +32,9 @@ import okhttp3.Protocol
 import okhttp3.Request
 import okhttp3.Response
 import org.assertj.core.api.Assertions.assertThat
+import org.junit.After
 import org.junit.Assert.fail
 import org.junit.Before
-import org.junit.Ignore
```
Comment (oldergod) [inline wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt]: You still want it.

**Comment by oldergod** [comment]
Thank you!

File: wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt (hunk 5 above)
Comment (swankjesse) [reply wire-grpc-tests/src/test/java/com/squareup/wire/GrpcClientTest.kt]: indeed

**Comment by swankjesse** [comment]
Looks like I didn’t fix it all. I’m looking.

```

        Caused by:
        kotlinx.coroutines.channels.ClosedReceiveChannelException: Channel was closed
            at kotlinx.coroutines.channels.Closed.getReceiveException(AbstractChannel.kt:1094)
            at kotlinx.coroutines.channels.AbstractChannel.receiveResult(AbstractChannel.kt:586)
            at kotlinx.coroutines.channels.AbstractChannel.receive(AbstractChannel.kt:579)
            at com.squareup.wire.GrpcClientTest$streamingResponseSuspend$1.invokeSuspend(GrpcClientTest.kt:246)
            at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
            at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:238)
            at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.kt:116)
            at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:80)
            at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:54)
            at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
            at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:36)
            at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
            at com.squareup.wire.GrpcClientTest.streamingResponseSuspend(GrpcClientTest.kt:242)

```

File: wire-grpc-client/src/main/java/com/squareup/wire/internal/RealGrpcStreamingCall.kt [hunk 6]
```diff
@@ -53,9 +53,11 @@ internal class RealGrpcStreamingCall<S : Any, R : Any>(
...
 
     responseChannel.invokeOnClose {
-      call.cancel()
-      requestChannel.cancel()
-      responseChannel.cancel()
+      if (responseChannel.isClosedForReceive) {
+        // Short-circuit the request stream if it's still active.
+        call.cancel()
+        requestChannel.cancel()
+      }
```
Comment (swankjesse) [inline wire-grpc-client/src/main/java/com/squareup/wire/internal/RealGrpcStreamingCall.kt]: FYI @oldergod @Egorand I had to add the if check here too. This gets called when the stream is out of messages (not a cancel) and when the reader expresses disinterest (a cancel). We only want to break the request channel on cancels.

**Pull request merged**